compose/ui/ui-text/src/desktopMain/kotlin/androidx/compose/ui/text/platform/AwtFontInterop.kt

```diff
@@ -0,0 +1,161 @@
+ * accessible, you should pass this argument to the JVM: `--add-opens
+ * java.desktop/sun.font=ALL-UNNAMED`.
+ *
+ * @return The resolved physical font name, or `null` if it can't be
+ *     resolved (either it's unknown, or [isAbleToResolveFontProperties] is
+ *     false)
+ */
+@ExperimentalTextApi
+fun Font.resolveFontFamilyNameOrNull(): String? =
+    AwtFontUtils.resolvePhysicalFontFamilyNameOrNull(fontFamilyNameOrNull() ?: family, style)
+
+/**
+ * Load a [FontFamily] from a font family name. The Compose runtime is able
+ * to load font families available in the system, and ones embedded in the
+ * JetBrains Runtime (see note below).
+ *
+ * If a family is not known to Compose, the lookup will return a fallback
+ * font family.
+ *
+ * If you're trying to use an AWT [Font] in Compose, use the
+ * [Font.asFontFamily] function instead, which will take care of some
+ * AWT-specific quirks, too.
+ *
+ * **Note:** when running on Java VMs other than the JetBrains Runtime,
+ * only system fonts are available. Embedded JVM fonts on other runtimes
+ * are not supported.
+ *
+ * @param familyName The name of the font family to load
+ * @param onlySystemFonts If true, embedded JBR fonts are ignored
+ * @see Font.asFontFamily
+ */
+@ExperimentalTextApi
+@Stable
+fun FontFamily(familyName: String, onlySystemFonts: Boolean = false): FontFamily {
+    // First, try loading from embedded JetBrains Runtime fonts (if any)
+    val embeddedFamily = JetBrainsRuntimeFontFamilies.embeddedFamilies[familyName.lowercase()]
+    if (!onlySystemFonts && embeddedFamily != null) return embeddedFamily
+
+    // Second, try loading as system font family (will fall back to defaults if needed)
+    return FontFamily(
+        SystemFont(familyName, FontWeight.W100, FontStyle.Normal),
+        SystemFont(familyName, FontWeight.W200, FontStyle.Normal),
+        SystemFont(familyName, FontWeight.W300, FontStyle.Normal),
+        SystemFont(familyName, FontWeight.W400, FontStyle.Normal),
+        SystemFont(familyName, FontWeight.W500, FontStyle.Normal),
+        SystemFont(familyName, FontWeight.W600, FontStyle.Normal),
+        SystemFont(familyName, FontWeight.W700, FontStyle.Normal),
+        SystemFont(familyName, FontWeight.W800, FontStyle.Normal),
+        SystemFont(familyName, FontWeight.W900, FontStyle.Normal),
+        SystemFont(familyName, FontWeight.W100, FontStyle.Italic),
+        SystemFont(familyName, FontWeight.W200, FontStyle.Italic),
+        SystemFont(familyName, FontWeight.W300, FontStyle.Italic),
+        SystemFont(familyName, FontWeight.W400, FontStyle.Italic),
+        SystemFont(familyName, FontWeight.W500, FontStyle.Italic),
+        SystemFont(familyName, FontWeight.W600, FontStyle.Italic),
+        SystemFont(familyName, FontWeight.W700, FontStyle.Italic),
+        SystemFont(familyName, FontWeight.W800, FontStyle.Italic),
+        SystemFont(familyName, FontWeight.W900, FontStyle.Italic),
```

Comment:
This is necessary, because otherwise it doesn't pick up weights/styles. It will fall back to existing fonts if it can't actually find the right one. Example:

<img width="684" alt="image" src="https://github.com/JetBrains/compose-multiplatform-core/assets/153802/cfa7fa39-5b42-40e9-bb96-530fad4f8506">

Reply:
Can we wrap it into a loop here?

Reply:
We can, but it's slower, and we know in advance all the permutations won't change